tests/integration/dynamic_ip_test.py

```diff
@@ -577,6 +578,117 @@ def test_ipv6_autoconf_only(dhcpcli_up):
     libnmstate.apply(desired_state)
 
 
+def test_static_ipv4_with_routes_swich_back_to_dhcp(dhcpcli_up):
+    desired_state = dhcpcli_up
+    dhcp_cli_desired_state = desired_state[INTERFACES][0]
+    dhcp_cli_desired_state[Interface.STATE] = InterfaceState.UP
+    dhcp_cli_desired_state[Interface.IPV4] = create_ipv4_state(
+        enabled=True, dhcp=False
+    )
+    dhcp_cli_desired_state[Interface.IPV4][InterfaceIPv4.ADDRESS] = [
+        {
+            InterfaceIPv4.ADDRESS_IP: IPV4_ADDRESS2,
+            InterfaceIPv4.ADDRESS_PREFIX_LENGTH: 24,
+        }
+    ]
+
+    desired_state[RT.KEY] = {
+        RT.CONFIG: [
+            {
+                RT.DESTINATION: IPV4_DEFAULT_GATEWAY,
+                RT.NEXT_HOP_ADDRESS: DHCP_SRV_IP4,
+                RT.NEXT_HOP_INTERFACE: DHCP_CLI_NIC,
+            },
+            {
+                RT.DESTINATION: '203.0.113.0/24',
+                RT.NEXT_HOP_ADDRESS: DHCP_SRV_IP4,
+                RT.NEXT_HOP_INTERFACE: DHCP_CLI_NIC,
+            },
+        ]
+    }
+
+    libnmstate.apply(desired_state)
+    assertlib.assert_state(desired_state)
+
+    desired_state = dhcpcli_up
```

Comment:
Does this really do something? This is already happening right after the function definition. Since there is no deepcopy happening, everything is operating on the same dictionary - or am I missing something?

Reply:
Removed.